### PR TITLE
Format `if` statements

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

This PR implements the formatting of `if` statements. I haven't verified if it covers all edge cases because
that will be easier once we have better coverage (you can simply go througth the black differences), but I believe it covers most reasonable uses pretty well. 

## Test Plan

A few own unit tests and reviewed (as best as I can) the black differences

### CODE REVIEW

**Comment by MichaReiser** [comment]
Current dependencies on/for this PR:
* main
  * **PR #4954** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/4954" target="_blank">Graphite</a> 
    * **PR #4921** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/4921" target="_blank">Graphite</a> 
      * **PR #4922** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/4922" target="_blank">Graphite</a> 
        * **PR #4951** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/4951" target="_blank">Graphite</a> 
          * **PR #4961** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/4961" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/astral-sh/ruff/4961?utm_source=stack-comment).

File: crates/ruff_python_formatter/src/snapshots/ruff_python_formatter__tests__ruff_test__statement__if_statement_py.snap [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+else:
+    ...
+```
+
+
+
+## Output
+```py
+if NOT_IMPLEMENTED_left < NOT_IMPLEMENTED_right:  # trailing if condition
```
Comment (T-256) [inline crates/ruff_python_formatter/src/snapshots/ruff_python_formatter__tests__ruff_test__statement__if_statement_py.snap:50]: here it should use `==` instead `<`

File: crates/ruff_python_formatter/src/statement/stmt_if.rs [hunk 2]
```diff
@@ -1,12 +1,127 @@
...
+                    text(":"),
+                    trailing_comments(if_trailing_comments),
+                    block_indent(&body.format())
+                ]
+            )?;
+
+            else_comments = trailing_alternate_comments;
+            last_node_of_previous_body = body.last();
+
+            if let Some(elif) = else_if(orelse) {
```
Comment (konstin) [inline crates/ruff_python_formatter/src/statement/stmt_if.rs:61]: Does this mean there's no elif node, just if?

**Review by konstin — APPROVED** [review]

File: crates/ruff_python_formatter/src/statement/stmt_if.rs (hunk 2 above)
Comment (MichaReiser) [reply crates/ruff_python_formatter/src/statement/stmt_if.rs:61]: Yep. `elif` is represented as an `else` body that contains a single `if` statement. Rome (and Roslyn) model this very similarly where the `alternate` (else) is either an `if` statement (in which it is an `else if`) or a block (an else).

File: crates/ruff_python_formatter/src/lib.rs [hunk 3]
```diff
@@ -436,6 +437,11 @@ def with_leading_comment(): ...
         //     .document()
         //     .display(formatted.context().source_code()));
 
+        // dbg!(formatted
+        //     .context()
+        //     .comments()
+        //     .debug(formatted.context().source_code()));
+
```
[the remark this answers is not in the document]
Comment (MichaReiser) [reply crates/ruff_python_formatter/src/lib.rs:444]: I would prefer not removing the code. I keep typing this every now and then when debugging comments

[the remark this answers is not in the document]
Comment (konstin) [reply crates/ruff_python_formatter/src/lib.rs:444]: What about
```rust
trait DebugComments {
    /// Used for debugging the comment implementation
    #[allow(unused)]
    fn debug_comments(&self) -> DisplayDocument;
}

impl<'a> DebugComments for Formatted<PyFormatContext<'a>> {
    fn debug_comments(&self) -> DisplayDocument {
        self.document().display(self.context().source_code())
    }
}
```

**Pull request merged**